script/tool/lib/src/format_command.dart

```diff
@@ -342,6 +359,13 @@ class FormatCommand extends PackageCommand {
         (String filePath) => extensions.contains(path.extension(filePath)));
   }
 
+  Iterable<String> _filterGeneratedFiles(Iterable<String> files) {
```

Comment:
Rather than filtering out generated files, we just need Pigeon to do Swift formatting in CI; [this code](https://github.com/flutter/packages/blob/129e08c1dff159b1b49484e1f838f0a471273cd7/packages/pigeon/tool/shared/generation.dart#L239-L246) should include Swift formatting. If we changed the default then it'll just work, otherwise we need to add `--swift` there.

Reply:
Oh, but that's not going to work on the Linux machine where we run this.

Hm, do we have a `clang-format` CIPD package for macOS already? If so, we could change [this](https://github.com/flutter/packages/blob/129e08c1dff159b1b49484e1f838f0a471273cd7/packages/pigeon/tool/run_tests.dart#L211) to macOS instead of Linux, and adjust the dependencies in ci.yaml. If not I'll need to think more about how we can handles this sanely, and we can land with the exclusion in the meantime.

Reply:
Maybe it's easier to change the Pigeon validator to check specific languages for changes instead of checking everything for changes, and then run it for different languages on different CI hosts the way we do the `format` check itself. I'll play with that today.

Reply:
> Maybe it's easier to change the Pigeon validator to check specific languages for changes instead of checking everything for changes, and then run it for different languages on different CI hosts the way we do the `format` check itself.

I think https://github.com/flutter/packages/pull/5944 should make this work.

Reply:
I changed my comment about removing it into a TODO.  How about I land this, then you rebase https://github.com/flutter/packages/pull/5944 onto ToT and remove the `_filterGeneratedFiles` in that PR?

Reply:
👍🏻

Reply:
> Rather than filtering out generated files, we just need Pigeon to do Swift formatting in CI; [this code](https://github.com/flutter/packages/blob/129e08c1dff159b1b49484e1f838f0a471273cd7/packages/pigeon/tool/shared/generation.dart#L239-L246) should include Swift formatting.

Oops, now that `swift` is enabled by default, that same code will need (until my PR lands) `--no-swift`.